test/gtest/uct/uct_test.cc

```diff
@@ -1470,6 +1470,48 @@ ucs_status_t uct_test::send_am_message(entity *e, uint8_t am_id, int ep_idx)
     }
 }
 
+void uct_test::async_event_ctx::signal() {
+    ASSERT_TRUE(aux_pipe_init);
+    ucs_async_pipe_push(&aux_pipe);
+}
+
+bool uct_test::async_event_ctx::check_event(entity &e, int timeout) {
+    if (ucs_unlikely(wakeup_fd.fd == -1)) {
+        /* create wakeup */
+        if (e.iface_attr().cap.event_flags & UCT_IFACE_FLAG_EVENT_FD) {
+            ucs_status_t status =
+                uct_iface_event_fd_get(e.iface(), &wakeup_fd.fd);
+            ASSERT_UCS_OK(status);
+        } else {
+            ucs_status_t status =
+                ucs_async_pipe_create(&aux_pipe);
+            ASSERT_UCS_OK(status);
+            aux_pipe_init = true;
+            wakeup_fd.fd = ucs_async_pipe_rfd(&aux_pipe);
+        }
+    }
+
+    int ret = poll(&wakeup_fd, 1, timeout);
+    EXPECT_TRUE((ret == 0) || (ret == 1));
+    if (ret > 0) {
+        if (e.iface_attr().cap.event_flags &
+            UCT_IFACE_FLAG_EVENT_ASYNC_CB) {
+            ucs_async_pipe_drain(&aux_pipe);
+        }
+        return true;
+    }
+
+    return false;
+}
+
+void uct_test::async_event_ctx::reset() {
+    if (aux_pipe_init) {
```

Comment:
maybe check if aux_pipe!=-1?

Reply:
aux_pipe is a structure of two read/write fds